Doc/using/cmdline.rst

```diff
@@ -907,6 +913,28 @@ conflict.
 
    .. versionadded:: 3.7
 
+.. envvar:: PYTHONWARNDEFAULTENCODING
+
+   If this environment variable is set to a non-empty string, issue a
+   :class:`EncodingWarning` when an ``encoding`` option is omitted and
+   the default encoding is locale-specific.
+
+   This option can be used to find bugs caused by not passing
+   ``encoding="utf8"`` option. For example::
+
+      # This code may cause UnicodeDecodeError on Windows.
+      # encoding="utf8" or "b" mode must be used.
+      with open(path) as f:
+          data = json.load(f)
+
+   ``encoding="locale"`` option can be used to specify locale-specific
+   encoding explicitly since Python 3.10. Python won't issue a
+   :class:`EncodingWarning` for it.
```

Comment:
The os documentation contains multiple sections about encodings. It sounds like the natural home for documentation about encodings used to read/write text files. What do you think about adding a new section there? See for example https://docs.python.org/dev/library/os.html#python-utf-8-mode 

It avoids to repeat the document at the env var, -X option, PyConfig doc, etc. Example:
https://docs.python.org/dev/using/cmdline.html#envvar-PYTHONUTF8

Reply:
I added a new section in library/io.rst.